### PR TITLE
Switch to using mb_(encode|decode)_mimeheader functions

### PR DESCRIPTION
**Pull request by Pudge601**
The new "Fixed" section in the change log shows the issues that this change will fix.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -25,7 +25,8 @@
     "require": {
         "php": ">=5.6",
         "ext-iconv" : "*",
```
Comment (chrisminett) [inline composer.json]: Can we drop requirement for *iconv* now?

File: tests/__files/attachments-expected-headers.txt [hunk 2]
```diff
@@ -3,20 +3,51 @@ From: From Name <from@example.com>
...
+ =?UTF-8?B?TnhibjRteTJSWkx2OVNTS1lSc05yK1NPTVBzRUFqTkpKR29XYWNFNy9KbVc3?=
+ =?UTF-8?B?aVZDV3BHQjBjbzdFanhocjNFd1VNMCAgICAgICAgIEcyZlpCNy9jUXJWN3pZ?=
+ =?UTF-8?B?SXJra29ldFJXWVRxVHZPdDdXOGxmRUphTFhGT1NBVHFXL1hjYW9zNUJXbzg4?=
+ =?UTF-8?B?ckpJbURXcmV3ICAgICAgICAgMWszWWJuTnMwanlYdlBPK2p5dFVmV0VrRFB1?=
+ =?UTF-8?B?N3cxaytLOVRxdkh0R2Vhd3lqMjFRZU5tbzFaMVAvL2cyOU1PNjFtL04gICAg?=
+ =?UTF-8?B?ICAgICBiVStJZXhkT0cvTzRYY2F1VTFRazhnR20weEEzc3pHWlhHYWFqaThl?=
+ =?UTF-8?B?Qmdrblk4RTZieE5JdElpRGFKOXZIR0x2eU1aaiAgICAgICAgIDZTR2c9PQ==?=
 X-Goomoji-Body: true
 Date: Thu, 16 Aug 2012 15:45:32 +0100
-Message-Id: <CAJ_TRnL=YMObf2FT9bU7NO0ziPYxpnxfxrtOz4r2-aBxkHSOrA@mail.gmail.com>
```
Comment (chrisminett) [inline tests/__files/attachments-expected-headers.txt:22]: This is a shame. Message-Id would be nice if it were readable.

File: composer.json (hunk 1 above)
Comment (Pudge601) [reply composer.json]: Looks like it, a quick global search for 'iconv' shows up nothing, which I believe means we can get rid of the requirement.

I'll remove this now.

File: tests/__files/attachments-expected-headers.txt (hunk 2 above)
Comment (Pudge601) [reply tests/__files/attachments-expected-headers.txt:22]: Yeah, I noticed this while I was updating these expected outputs..

It is a shame, but not sure what we can do about it to be honest :/

File: composer.json (hunk 1 above)
Comment (Pudge601) [reply composer.json]: I've made this change.

Do you think this needs an entry in the change log?

**Review by chrisminett — APPROVED** [review]

**Review by qubit05 — APPROVED** [review]

**Pull request merged**